src/Storages/FileLog/FileLogDirectoryWatcher.cpp

```diff
@@ -0,0 +1,88 @@
+#include <Storages/FileLog/FileLogDirectoryWatcher.h>
+#include <Poco/Delegate.h>
+#include <Poco/DirectoryWatcher.h>
+
+FileLogDirectoryWatcher::FileLogDirectoryWatcher(const std::string & path_)
+    : path(path_), dw(std::make_shared<Poco::DirectoryWatcher>(path))
+{
+    dw->itemAdded += Poco::delegate(this, &FileLogDirectoryWatcher::onItemAdded);
+    dw->itemRemoved += Poco::delegate(this, &FileLogDirectoryWatcher::onItemRemoved);
+    dw->itemModified += Poco::delegate(this, &FileLogDirectoryWatcher::onItemModified);
+    dw->itemMovedFrom += Poco::delegate(this, &FileLogDirectoryWatcher::onItemMovedFrom);
+    dw->itemMovedTo += Poco::delegate(this, &FileLogDirectoryWatcher::onItemMovedTo);
+}
+
+FileLogDirectoryWatcher::Events FileLogDirectoryWatcher::getEvents()
+{
+    std::lock_guard<std::mutex> lock(mutex);
+    Events res;
+    res.swap(events);
+    return res;
```

Comment:
Then may be rename the method? becuase calling `getEvents` you do not expect that events will be cleaned.
Or return a copy instead of reference, I do not think it is bad in this case.

Reply:
Ok.